frontend/src/tests/user-permission-tests.js

```diff
@@ -14,11 +14,41 @@ import { databaseSeeder } from "./setup";
 export function userPermissionsTests(api) {
     const { apiGET, apiPOST } = api;
 
-    const newUserData = {
+    const newInstructorData = {
         utorid: "userxx",
         roles: ["instructor"],
     };
 
+    const newTAData = {
+        utorid: "tauser",
+        roles: ["ta"],
+    };
+
+    const newTAInstructorData = {
+        utorid: "tainstuser",
+        roles: ["ta", "instructor"],
+    };
+
+    async function runTestWithUser(userData, testBody) {
```

Comment:
Need to parse this function into a setup-teardown style instead of passing a function into a function